src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/gcs/gcs_server/gcs_task_manager.h"
+
+#include "ray/common/ray_config.h"
+#include "ray/common/status.h"
+
+namespace ray {
+namespace gcs {
+
+void GcsTaskManager::Stop() {
+  io_service_.stop();
+  if (io_service_thread_->joinable()) {
+    io_service_thread_->join();
+  }
+}
+
+absl::optional<rpc::TaskEvents>
```

Comment:
Consider returning bool? I think the dropped counter can be tracked within storage?

Reply:
I feel the drop count could actually be tracked outside of the storage, such that the storage could be better decoupled. But open to move the tracking in.